recipes/sioclient/all/conanfile.py

```diff
@@ -0,0 +1,116 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import copy, get, replace_in_file, rm, rmdir, save
+
+required_conan_version = ">=1.53.0"
+
+
+class SioclientConan(ConanFile):
+    name = "sioclient"
+    description = "C++11 implementation of Socket.IO client"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/socketio/socket.io-client-cpp"
+    topics = ("websocket", "client")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("websocketpp/0.8.2")
+        self.requires("asio/1.30.2")
+        self.requires("rapidjson/cci.20230929")
+        if self.options.with_openssl:
+            self.requires("openssl/[>=1.1 <4]")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 11)
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Shared builds on Windows are not supported")
+
+    def build_requirements(self):
+        # 3.28+ is not supported
+        self.tool_requires("cmake/[>=3.12 <=3.27]")
```

Comment:
I'll need to check with the team what to do to set a min cmake version like this. (Also, this probably should be set to something >= 3.15)